### ISSUE TITLE
JSDoc plugins or templates cannot be reached when JSDoc-cing a project which has plugins / templates for its own purposes

### ISSUE THREAD

**Issue by GerHobbelt** [jsdoc3/jsdoc#1081]
example jsdoc.conf.json:

```
{
    "plugins": [
        "plugins/markdown",
        "plugins/summarize"
    ],
    "opts": {
        "template": "templates/default",
    }
}
```

will fail to find those plugins when running `jsdoc -c jsdoc.conf.json` in a project directory which itself also has its own plugins sitting in a `plugins/\<filename\>` subtree, as JSDoc first looks in the current directory (plus `plugins`) but only checks the _directory_ part of the path to announce a find internally.

Ditto for templates: when you have your own `templates/default/\<whatever.js>` in your project, JSDoc will get thwarted by that one.

Pullreq with a fix for this coming up....

**Comment by hegemonic** [comment]
Sorry for the delay in looking at your patch. It looks correct to me, so I've merged it into master. The fix will be included in JSDoc 3.4.1.

Thanks for the fix!

**Comment by GerHobbelt** [comment]
Super!

Met vriendelijke groeten / Best regards,

Ger Hobbelt

--------------------------------------------------
web:    http://www.hobbelt.com/
        http://www.hebbut.net/
mail:   ger@hobbelt.com
mobile: +31-6-11 120 978
--------------------------------------------------

**Comment by hegemonic** [comment]
Reopening because the fix caused #1259. I'm reverting the fix on master and releases/3.4.

**Comment by hegemonic** [comment]
Fixed (again! hopefully!) on master. The fix will be included in JSDoc 3.5.0.
